### ISSUE TITLE
1st draft version

### ISSUE THREAD

**Issue by TermeHansen** [TermeHansen/home-assistant-drtv-play#1]
I will go through https://github.com/lindell/home-assistant-svt-play and make a 1st draft version for drtv to be tested...

**Comment by agdamsbo** [comment]
You are a hero! I would imagine this could work for any media player entity with a screen. I have a touch panel with BrowserMod, that I can stream video, so it would not be limited to Chromecast.

I have been through the [https://github.com/lindell/home-assistant-svt-play](https://github.com/lindell/home-assistant-svt-play) as well. As far as I can gather, they are stitching a url together based on name fragments (episode, series etc). It doesn't implement a media browser function, like I would imagine your tvapi, would be able to provide.

**Comment by TermeHansen** [comment]
Have you found any other media sources for HA that we should be inspired from?

**Comment by agdamsbo** [comment]
I have been looking at the [Radio Browser](https://github.com/home-assistant/core/tree/dev/homeassistant/components/radio_browser), which arguably is only audio, but the browsing function is very basic. A minimum goal for this could be to get "Programmer A-Å" working or something similar. Saving search, favorites and subtitles for later.

**Comment by agdamsbo** [comment]
There is also the [Kodi](https://github.com/home-assistant/core/tree/dev/homeassistant/components/kodi) and [dlna_dms](https://github.com/home-assistant/core/tree/dev/homeassistant/components/dlna_dms) providing mixed medias as source, though these are for local media, which I would suspect works a little different than streaming web links.

**Comment by agdamsbo** [comment]
This [link on svtplay-dl in the HA Forum](https://community.home-assistant.io/t/svt-play-and-chromecast/72713/17), was what got me hooked on that there might be a way. I curently have a helper entity in HA to input the stream path and a script to stream the provided path. I think that is very much like the home-assistant-svt-play-approach. I see they have [been discussing the browsing function as well](https://github.com/lindell/home-assistant-svt-play/issues/4), though they are not approaching it for now.

As I see you @TermeHansen  have been reverse engineering the DRTV api, I'd suspect the same work would be necessary for the same approach to work with SVT (they do actually have a GitHub rep with a [Android Streaming App](https://github.com/svt/android-media-streaming)). Including other sources like the svtplay-dl has would open the use base and possible help.

**Comment by TermeHansen** [comment]
Still trying to understand the smartness factor here. Doesn't Chromecast take links shared directly from the drtv app? Why would that flow be nicer via an "app" in HA?

**Comment by agdamsbo** [comment]
Fair point. My phone doesn't support chromecast. That's what got me interested in other approaches. I thought that with all the work you already had put in the Kodi-addon, the proting would be possible, without having to rewrite everything. I might very well be wrong.

Being able to use HA for automations (start the latest favourite children's program, when turning on the tv in the morning (yes, I have kids)), notify on new episodes, start playing from wall tablet would help limit the use of phones in the house (which is a goal for me, to limit other distractions). Having different NFC tags and [a scanner](https://github.com/adonno/tagreader) would give an "analogue" interface to start specific programs (I find that we mostly watch the same 3-4 programs). All these use cases would be possible with the "lindell-approach", but being able to browse would just make it much easier and approachable.

Lately I have found the [Music Assistant](https://github.com/music-assistant) integration (currently with v2 in beta), which aims to connect audio and speakers from different sources and makes for an integrated, synchronised, multispeaker, experience, and is really working well. Being able to also control video playback from HA would really be amazing (Jellyfin, DRTV, dlna and other sources).

**Comment by TermeHansen** [comment]
OK I have made a 1st minimal test, where only `play_channel` works
![image](https://github.com/TermeHansen/home-assistant-drtv-play/assets/6922018/dcf76771-7cad-449d-8234-ba5be7404dc1)
it should work for the channels 
`dr1` 
`dr2` 
`drtv`
`dr ramasjang`

you can test it if you want in https://github.com/TermeHansen/home-assistant-drtv-play/tree/drafting

I need to learn a bit more on how HA works the best with the async calls and how much of the tvapi from the kodi addon I potentially should update for it.

**Comment by agdamsbo** [comment]
This is great. Thank you for taking up the challenge!
For the life of me, I cant get it to install on HA. I'll keep trying.
I will try to look into your code and see if I can contribute, but I might me more help in documenting.

**Comment by TermeHansen** [comment]
You need to add the `drtv_play` directory from my branch in the `config/custom_components` folder and add `drtv_play:` to your `config/configuration.yaml`

Right now here on my install:
```
$ tail config/configuration.yaml 
svt_play:
drtv_play:

$ ls config/custom_components/
drtv_play  svt_play

```

**Comment by agdamsbo** [comment]
I missed the `drtv_play:`in `/configuration.yaml`

Now I get the error `Component error: drtv_play - No module named 'requests_cache'`, when running "Check configuration" in Developer Tools before restarting.

**Comment by TermeHansen** [comment]
sorry forgot to add this here on github, https://github.com/TermeHansen/home-assistant-drtv-play/commit/e4760dc6f572db713546500cd9994a6392207f9d

just add to your local `manifest.json`

https://github.com/TermeHansen/home-assistant-drtv-play/blob/e4760dc6f572db713546500cd9994a6392207f9d/custom_components/drtv_play/manifest.json#L13

**Comment by TermeHansen** [comment]
I think I will follow the code of the Jellyfin integration for how to browse the media
https://github.com/home-assistant/core/tree/dev/homeassistant/components/jellyfin
Have you tried it? Anything that could be good or bad for our use with drtv?

**Comment by agdamsbo** [comment]
I realise that I am not able to browse media in the Jellyfin plugin. So it might not be that good of an example. The [DLNA server addon](https://github.com/home-assistant/core/tree/dev/homeassistant/components/dlna_dms) is working very well though when browsing media. It is just a very simple folder tree structure. No search or anything.

**Comment by agdamsbo** [comment]
And the draft addon is now working!

**Comment by TermeHansen** [comment]
@agdamsbo I think we should continue the discussion and testing of this 1st drat in the PR  https://github.com/TermeHansen/home-assistant-drtv-play/pull/2
